router.go

```diff
@@ -160,6 +161,16 @@ func treePath(method, path string) string {
 	return "/" + method + path
 }
 
+//validMethod is copied from net/http/request.go
+func validMethod(method string) bool {
+	return len(method) > 0 && strings.IndexFunc(method, isNotToken) == -1
+}
+
+//isNotToken is copied from net/http/http.go
+func isNotToken(r rune) bool {
```

Comment:
Same as above. At least some comment like "tests for HTTP methods."

Reply:
added referents to validMethod in the comment